controlplane/kubeadm/internal/controllers/upgrade.go

```diff
@@ -74,6 +74,16 @@ func (r *KubeadmControlPlaneReconciler) upgradeControlPlane(
 		return ctrl.Result{}, errors.Wrap(err, "failed to update the kubernetes version in the kubeadm config map")
 	}
 
+	// FIXME(chrischdi): use a better semver comparison
```

Comment:
😆 I think version.ParseMajorMinorPatchTolerant (with or without tolerant) is the best we've got, but there's a few patterns in the codebase.

Reply:
And you can use the version.Compare to get the comparison instead of directly comparing the minor versions

Reply:
The latter is what the fixme is about :-)

Reply:
Independent of how we implement it, please ensure that alpha/beta/... versions are also covered :)

Reply:
yes, will have to because the test will otherwise fail (due to k/k master getting an v1.25 alpha version in the test)